capirca/lib/nsxt.py

```diff
@@ -0,0 +1,320 @@
+
+    for header, terms in pol.filters:
+      if self._PLATFORM not in header.platforms:
+        continue
+
+      filter_options = header.FilterOptions(self._PLATFORM)
+      if len(filter_options) >= 2:
+        filter_name = filter_options[1]
+
+      self._ParseFilterOptions(filter_options)
+
+      filter_type = ''
+      applied_to = ''
+
+      term_names = set()
+      new_terms = []
+      for term in terms:
+        # Check for duplicate terms
+        if term.name in term_names:
+          raise NsxtDuplicateTermError('There are multiple terms named: %s' %
+                                       term.name)
+        term_names.add(term.name)
+
+        term.name = self.FixTermLength(term.name)
+
+        new_terms.append(Term(term, filter_type, applied_to, 4))
+
+      self.nsxt_policies.append((header, filter_name, [filter_type],
+                                 new_terms))
+
+  def _ParseFilterOptions(self, filter_options):
+    """Parses the target in header for filter type, section_id and applied_to.
+
+    Args:
+      filter_options: list of remaining target options
+
+    Returns:
+      A dictionary that contains fields necessary to create the firewall
+      rule.
+
+    Raises:
+      UnsupportedNsxtAccessListError: Raised when we're give a non named access
+      list.
+    """
+    # check for filter type
+    if not 2 <= len(filter_options) <= 5:
+      raise UnsupportedNsxtAccessListError(
+          'Invalid Number of options specified: %d. Required options '
+          'are: filter type and section name. Platform: %s' % (
+              len(filter_options), self._PLATFORM))
+    # mandatory section_name
+    section_name = filter_options[0]
+    # mandatory
+    filter_type = filter_options[1]
+
+    # a mixed filter outputs both ipv4 and ipv6 acls in the same output file
+    good_filters = ['inet', 'inet6', 'mixed']
+
+    # check if filter type is renderable
+    if filter_type not in good_filters:
+      raise UnsupportedNsxtAccessListError(
+          'Access list type %s not supported by %s (good types: %s)' % (
+              filter_type, self._PLATFORM, str(good_filters)))
+
+    section_id = 0
+    applied_to = None
+    filter_opt_len = len(filter_options)
+
+    if filter_opt_len > 2:
+      for index in range(2, filter_opt_len):
+        if index == 2 and filter_options[2] != 'securitygroup':
+          section_id = filter_options[2]
+          continue
+        if filter_options[index] == 'securitygroup':
+          if index + 1 <= filter_opt_len - 1:
+            applied_to = filter_options[index + 1]
+            break
+          else:
+            raise UnsupportedNsxtAccessListError(
+                'Security Group Id is not provided for %s' % (self._PLATFORM))
+
+    self._FILTER_OPTIONS_DICT['section_name'] = section_name
+    self._FILTER_OPTIONS_DICT['filter_type'] = filter_type
+    self._FILTER_OPTIONS_DICT['section_id'] = section_id
+    self._FILTER_OPTIONS_DICT['applied_to'] = applied_to
```

Comment:
Looks like applied_to -- the ID of the security group -- does not get used anywhere?